.changeset/blue-moons-travel.md

```diff
@@ -0,0 +1,33 @@
+---
+"@comet/cms-api": minor
+---
+
+Set content scopes in request object
+
+This allows accessing the affected content scopes inside a block's transformer service.
+
+**Example**
+
+```ts
+import { Inject, Injectable } from "@nestjs/common";
+import { CONTEXT } from "@nestjs/graphql";
+
+/* ... */
+
+@Injectable()
+export class PixelImageBlockTransformerService implements BlockTransformerServiceInterface<PixelImageBlockData, TransformResponse> {
+    constructor(
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        @Inject(CONTEXT) private readonly context: any,
```

Comment:
We're using any for context everywhere, should we change that?

Reply:
better typing is always wanted, if possible. Is it possible here?

Reply:
I'm not sure what exactly context is here. It could be http.IncomingMessage or express' Request. Maybe I should change this everywhere in a follow-up PR?

Reply:
Did some digging: The context currently has the shape:

```ts
interface Context extends express.Request {
  req: http.IncomingMessage
}
```

This is because of our context function here: https://github.com/vivid-planet/comet/blob/content-scopes-in-execution-context/demo/api/src/app.module.ts#L77

The `req` passed to the context function is express' request (see [the docs](https://www.apollographql.com/docs/apollo-server/v3/api/apollo-server#middleware-specific-context-fields)). For some reason, `req` is added to the context as well (probably Nest adds this)

IMO this duplication is complicated and error-prone. We should investigate if it's even necessary to add such a context function at all. However, since this changes for Apollo Server v4, we could change this in next and then type context correctly. What do you think?

Reply:
interesting.

yes, that should be investigated (of course not in this PR)

We should also try to get independent of express in the library (to be able to switch to fastify)